src/system.jl

```diff
@@ -0,0 +1,302 @@
+"""
+struct SystemDA <: System
+    gens_per_bus::Dict{String, Vector{Int}}
+    incs_per_bus::Dict{String, Vector{String}}
+    decs_per_bus::Dict{String, Vector{String}}
+    psds_per_bus::Dict{String, Vector{String}}
+    loads_per_bus::Dict{String, Vector{String}}
+
+    zones::Vector{Zone}
+    buses::Buses
+    generators::Generators
+    branches::Branches
+    LODF::Dict{String, KeyedArray}
+    PTDF::KeyedArray
+
+    # Generator related time series
+    offer_curve::KeyedArray{Vector{Tuple{Float64, Float64}}}
+    availability::KeyedArray{Bool}
+    must_run::KeyedArray{Bool}
+    regulation_min::KeyedArray{Float64}
+    regulation_max::KeyedArray{Float64}
+    pmin::KeyedArray{Float64}
+    pmax::KeyedArray{Float64}
+    ancillary_services::ServicesTimeSeries
+
+    # Load time series
+    loads::KeyedArray{Float64}
+
+    # Virtuals/PSD time series
+    increment_bids::KeyedArray{Vector{Tuple{Float64, Float64}}}
+    decrement_bids::KeyedArray{Vector{Tuple{Float64, Float64}}}
+    price_sensitive_demand::KeyedArray{Vector{Tuple{Float64, Float64}}}
+end
+
+"""
+    struct SystemRT <: System
+
+Subtype of a `System` for modelling the real-time market.
+
+Fields:
+ - Topology
+     - `gens_per_bus::Dict{String, Vector{Int}}`
+     - `loads_per_bus::Dict{String, Vector{String}}`
+ - Static components
+     - `zones::Vector{Zone}`
+     - `buses::Buses`
+     - `generators::Generators`
+     - `branches::Branches`
+     - `LODF::Dict{String, KeyedArray}`
+     - `PTDF::KeyedArray`
+ - Time series
+     - `offer_curve::KeyedArray{Vector{Tuple{Float64, Float64}}}`
+     - `status::KeyedArray{Bool}`
+     - `status_regulation::KeyedArray{Bool}`
+     - `regulation_min::KeyedArray{Float64}`
+     - `regulation_max::KeyedArray{Float64}`
+     - `pmin::KeyedArray{Float64}`
+     - `pmax::KeyedArray{Float64}`
+     - `ancillary_services::ServicesTimeSeries`
+     - `loads::KeyedArray{Float64}`
+"""
+struct SystemRT <: System
+    gens_per_bus::Dict{String, Vector{Int}}
+    loads_per_bus::Dict{String, Vector{String}}
+
+    zones::Vector{Zone}
+    buses::Buses
+    generators::Generators
+    branches::Branches
+    LODF::Dict{String, KeyedArray}
+    PTDF::KeyedArray
+
+    # Generator related time series
+    offer_curve::KeyedArray{Vector{Tuple{Float64, Float64}}}
+    status::KeyedArray{Bool}
+    status_regulation::KeyedArray{Bool}
+    regulation_min::KeyedArray{Float64}
+    regulation_max::KeyedArray{Float64}
+    pmin::KeyedArray{Float64}
+    pmax::KeyedArray{Float64}
+    ancillary_services::ServicesTimeSeries
+
+    # Load time series
+    loads::KeyedArray{Float64}
+end
+
+function Base.show(io::IO, ::MIME"text/plain", system::T) where {T <: System}
+    Base.summary(io, system)
+    get(io, :compact, false) && return nothing
+    z = length(system.zones) - 1
+    print(io, " with $z Zones")
+    for c in [:buses, :generators, :branches]
+        l = length(getproperty(getproperty(system, c), :name))
```

Comment:
can this just be:
```suggestion
        l = length(getproperty(system, c))
```
?